.github/workflows/build_container_image.yaml

```
@@ -21,19 +21,15 @@ jobs:
       - name: Set branch name
         id: branch
         run: echo "::set-output name=name::$(BRANCH_NAME=${GITHUB_REF##*/}; echo ${BRANCH_NAME/release_/}-auto)"
-      - run: docker build . -t galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} -f .k8s_ci.Dockerfile
-      - run: docker tag galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} quay.io/galaxy-k8s/galaxy:${{ steps.branch.outputs.name }} && docker tag galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} quay.io/galaxy-k8s/galaxy:${{ steps.vars.outputs.sha_short }}
+      - run: docker build . -t galaxy/galaxy-min:${{ steps.branch.outputs.name }} -f .k8s_ci.Dockerfile
+      - run: docker tag galaxy/galaxy-min:${{ steps.branch.outputs.name }} quay.io/galaxy-min/galaxy:${{ steps.branch.outputs.name }}
       - name: Login to quay.io
         uses: actions-hub/docker/login@master
         env:
           DOCKER_USERNAME: ${{ secrets.QUAY_USERNAME }}
           DOCKER_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
           DOCKER_REGISTRY_URL: quay.io
-      - name: Push to quay.io with commit ID
-        uses: actions-hub/docker@master
-        with:
-          args: push quay.io/galaxy-k8s/galaxy:${{ steps.vars.outputs.sha_short }}
       - name: Push to quay.io with branch name
         uses: actions-hub/docker@master
         with:
-          args: push quay.io/galaxy-k8s/galaxy:${{ steps.branch.outputs.name }}
+          args: push quay.io/galaxy-min/galaxy:${{ steps.branch.outputs.name }}
```

Is there any extra metadata field where we could encode the commit sha in? I think that might be useful for debugging. Or maybe that's already handled in during the build ?

---

There's a file named GITREVISION that's baked into the image, that has the commit sha. I just realized that this is appending `-auto` to the branch name, which we had planned to remove. Will update in a sec.

---

@mvdbeek Perhaps this for metadata? https://github.com/galaxyproject/galaxy/pull/12837/commits/9ea44788ef70bda979a3d05cbc9b270d05baf33b
We could remove the GITREVISION file and just opt for the label, but it does have the advantage of storing the label regardless of whether you pass it in as a build arg or not. (Docker doesn't seem to support dynamic labels)

---

Yeah, that looks great, I'd keep both mechanisms.

---

Could we also add DockerHub back?

```suggestion
          args: push quay.io/galaxy-min/galaxy:${{ steps.branch.outputs.name }}
      - name: Login to DockerHub
        uses: actions-hub/docker/login@master
        env:
          DOCKER_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
      - name: Push to DockerHub with branch name
        uses: actions-hub/docker@master
        with:
          args: push galaxy/galaxy-min:${{ steps.branch.outputs.name }}
```